library/HTMLPurifier/Filter/ExtractStyleBlocks.php

```diff
@@ -62,12 +62,38 @@ public function __construct()
         $this->_class_attrdef = new HTMLPurifier_AttrDef_CSS_Ident();
         $this->_enum_attrdef = new HTMLPurifier_AttrDef_Enum(
             array(
+                'active',
+                'after',
+                'before',
+                'checked',
+                'disabled',
+                'empty',
+                'enabled',
                 'first-child',
+                'first-letter',
+                'first-line',
+                'first-of-type',
+                'focus',
+                'hover',
+                'in-range',
+                'invalid',
+                'last-child',
+                'last-of-type',
                 'link',
+                'nth-child(odd)',
+                'nth-child(even)',
+                'only-of-type',
+                'only-child',
+                'optional',
+                'out-of-range',
+                'read-only',
+                'read-write',
+                'required',
+                'root',
```

Comment:
This one is not OK by default, it lets you "escape" from the specified region where styling is occurred, assuming that an anchor was specified.